consensus/view_change.go

```diff
@@ -132,14 +134,14 @@ func (consensus *Consensus) startViewChange(viewID uint64) {
 	consensus.getLogger().Info().
 		Uint64("ViewChangingID", viewID).
 		Dur("timeoutDuration", duration).
-		Str("NextLeader", consensus.LeaderPubKey.SerializeToHexStr()).
+		Str("NextLeader", consensus.LeaderPubKey.Bytes.Hex()).
 		Msg("[startViewChange]")
 
-	for i, key := range consensus.PubKey.PublicKey {
-		if !consensus.IsValidatorInCommittee(key) {
+	for i, key := range consensus.PubKey {
+		if !consensus.IsValidatorInCommittee(key.Bytes) {
 			continue
 		}
-		msgToSend := consensus.constructViewChangeMessage(key, consensus.priKey.PrivateKey[i])
+		msgToSend := consensus.constructViewChangeMessage(key.Object, consensus.priKey.PrivateKey[i])
```

Comment:
can you pass `key.Bytes` instead of `key.Object` as internally it is serializing again?

Reply:
it's done in the next PR which refactor the private key.